diff --git a/collectors/proc.plugin/ipc.c b/collectors/proc.plugin/ipc.c
index 048fe74a..57f93727 100644
--- a/collectors/proc.plugin/ipc.c
+++ b/collectors/proc.plugin/ipc.c
@@ -2,10 +2,17 @@
 
 #include "plugin_proc.h"
 
+#ifdef __DragonFly__
+#define _KERNEL_STRUCTURES
+#define SEM_INFO 0
+#endif
 #include <sys/sem.h>
 #include <sys/msg.h>
 #include <sys/shm.h>
 
+#ifdef __DragonFly__
+#undef _KERNEL_STRUCTURES
+#endif
 
 #ifndef SEMVMX
 #define SEMVMX  32767  /* <= 32767 semaphore maximum value */
diff --git a/libnetdata/libnetdata.c b/libnetdata/libnetdata.c
index 095c38c2..b00870ea 100644
--- a/libnetdata/libnetdata.c
+++ b/libnetdata/libnetdata.c
@@ -5,7 +5,7 @@
 #ifdef __APPLE__
 #define INHERIT_NONE 0
 #endif /* __APPLE__ */
-#if defined(__FreeBSD__) || defined(__APPLE__)
+#if defined(__FreeBSD__) || defined(__DragonFly__) || defined(__APPLE__)
 #    define O_NOATIME     0
 #    define MADV_DONTFORK INHERIT_NONE
 #endif /* __FreeBSD__ || __APPLE__*/
@@ -1179,7 +1179,7 @@ int recursively_delete_dir(const char *path, const char *reason) {
 
 static int is_virtual_filesystem(const char *path, char **reason) {
 
-#if defined(__APPLE__) || defined(__FreeBSD__)
+#if defined(__APPLE__) || defined(__FreeBSD__) || defined(__DragonFly__) 
     (void)path;
     (void)reason;
 #else
diff --git a/libnetdata/libnetdata.h b/libnetdata/libnetdata.h
index 41642efd..1081aae8 100644
--- a/libnetdata/libnetdata.h
+++ b/libnetdata/libnetdata.h
@@ -30,15 +30,15 @@
 
 #else /* !defined(ENABLE_JEMALLOC) && !defined(ENABLE_TCMALLOC) */
 
-#if !(defined(__FreeBSD__) || defined(__APPLE__))
+#if !(defined(__FreeBSD__) || defined(__DragonFly__)) || defined(__APPLE__))
 #include <malloc.h>
-#endif /* __FreeBSD__ || __APPLE__ */
+#endif /* __FreeBSD__ || __DragonFly__ || __APPLE__ */
 
 #endif /* !defined(ENABLE_JEMALLOC) && !defined(ENABLE_TCMALLOC) */
 
 // ----------------------------------------------------------------------------
 
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD__) || defined(__DragonFly__))
 #include <pthread_np.h>
 #define NETDATA_OS_TYPE "freebsd"
 #elif defined(__APPLE__)
@@ -80,7 +80,11 @@
 #include <sys/un.h>
 #include <time.h>
 #include <unistd.h>
+#if defined(__DragonFly__)
+// #include <uuid/uuid.h>
+#else
 #include <uuid/uuid.h>
+#endif
 
 #ifdef HAVE_NETINET_IN_H
 #include <netinet/in.h>
diff --git a/libnetdata/threads/threads.c b/libnetdata/threads/threads.c
index 133d9a54..74bd1522 100644
--- a/libnetdata/threads/threads.c
+++ b/libnetdata/threads/threads.c
@@ -26,7 +26,7 @@ const char *netdata_thread_tag(void) {
 // compatibility library functions
 
 pid_t gettid(void) {
-#ifdef __FreeBSD__
+#ifdef defined(__FreeBSD__) || defined(__DragonFly__)
 
     return (pid_t)pthread_getthreadid_np();
 
@@ -44,7 +44,7 @@ pid_t gettid(void) {
 
     return (pid_t)syscall(SYS_gettid);
 
-#endif /* __FreeBSD__, __APPLE__*/
+#endif /* __FreeBSD__, __DragonFly__, __APPLE__*/
 }
 
 // ----------------------------------------------------------------------------
